app/screens/home/profile_screen.tsx

```diff
@@ -6,52 +6,87 @@ import { TopBar } from "@/components/navigation/TopBar";
 import { TouchableOpacity } from "react-native";
 import { Image } from "react-native";
 import * as ImagePicker from "expo-image-picker";
-import { Dimensions } from "react-native";
+import { Dimensions, StyleSheet } from "react-native";
 import { ThemedTextButton } from "@/components/theme/ThemedTextButton";
 import { Icon } from "react-native-elements";
 import FirestoreCtrl, { DBUser } from "@/firebase/FirestoreCtrl";
 import { logOut, resetEmail, resetPassword } from "@/types/Auth";
+import { ThemedTextInput } from "@/components/theme/ThemedTextInput";
 
 //TODO : change the colors for light mode
 const { width, height } = Dimensions.get("window");
 
 export default function ProfileScreen({
   user,
+  setUser,
   navigation,
   firestoreCtrl,
 }: {
   user: DBUser;
+  setUser: React.Dispatch<React.SetStateAction<DBUser | null>>;
   navigation: any;
   firestoreCtrl: FirestoreCtrl;
 }) {
-  const [isLoggedIn, setIsLoggedIn] = React.useState<Boolean>(
-    user ? true : false,
-  );
-  const [image, setImage] = React.useState<string | null>(null);
+  const userIsGuest = user.name === "Guest";
+
+  React.useEffect(() => {
+    const fetchProfilePicture = async () => {
+      const profilePicture = await firestoreCtrl.getProfilePicture(user.uid);
+      setImage(profilePicture || null);
+    };
+    fetchProfilePicture();
+  }, [user.uid]);
+
+  const [name, setName] = React.useState<string>(user.name);
+  
+  React.useEffect(() => {
+    setName(user.name);
+  }, [user.name]);
+
+  const [image, setImage] = React.useState<string | null>(user.image_id ? user.image_id : null);
 
   const pickImage = async () => {
-    console.log("Loading image");
-    let result = await ImagePicker.launchImageLibraryAsync({
-      mediaTypes: ImagePicker.MediaTypeOptions.All,
-      allowsEditing: true,
-      aspect: [4, 3],
-      quality: 1,
-    });
-    if (!result.canceled) {
-      setImage(result.assets[0].uri);
+    try {
+      let result = await ImagePicker.launchImageLibraryAsync({
```

Comment:
Maybe you could extract the ImagePicker in a const at the beginning of the file for better readability.

Reply:
Yes you're right I just did that 👍🏻